### PR TITLE
Remove "switch for efficiency on typical cases"

### PR DESCRIPTION
**Pull request by boy0001**
It's not an efficient tradeoff for how common it is.

Also change the BlockVector hashcode, because it was better before. idk
why you reverted it in the new vector system update.

### CODE REVIEW

**Comment by wizjany** [comment]
Wasn't so much reverted as the classes were rewritten. I wasn't part of this decision making process so I'm not sure if there's any reason for the hash code or the switch though.

**Comment by boy0001** [comment]
looks like me4502 was going to try adding object pooling at some point. Aside from specific cases, I wouldn't think that'd have much benefit either since most vectors are short lived and probably optimized away by the jvm.

The current hashcode is just a generic java approach of multiplying it by a prime number. Objectively it'd be marginally slower and result in more collisions.

File: worldedit-core/src/main/java/com/sk89q/worldedit/math/BlockVector3.java [hunk 1]
```diff
@@ -35,27 +35,16 @@
     public static final BlockVector3 UNIT_X = new BlockVector3(1, 0, 0);
     public static final BlockVector3 UNIT_Y = new BlockVector3(0, 1, 0);
     public static final BlockVector3 UNIT_Z = new BlockVector3(0, 0, 1);
+    public static final BlockVector3 UNIT_MINUS_X = new BlockVector3(-1, 0, 0);
+    public static final BlockVector3 UNIT_MINUS_Y = new BlockVector3(0, -1, 0);
+    public static final BlockVector3 UNIT_MINUS_Z = new BlockVector3(0, 0, -1);
```
Comment (wizjany) [inline worldedit-core/src/main/java/com/sk89q/worldedit/math/BlockVector3.java]: does it make sense to add the negative units to Vector as well?

Comment (boy0001) [reply worldedit-core/src/main/java/com/sk89q/worldedit/math/BlockVector3.java]: I didn't see anywhere need the other ones.

File: worldedit-core/src/main/java/com/sk89q/worldedit/math/BlockVector2.java [hunk 2]
```diff
@@ -60,18 +60,6 @@ public static BlockVector2 at(double x, double z) {
     }
 
     public static BlockVector2 at(int x, int z) {
-        switch (x) {
```
Comment (kenzierocks) [inline worldedit-core/src/main/java/com/sk89q/worldedit/math/BlockVector2.java]: This & the other switch removals should be reverted, performance tests show that this isn't any more efficient. The change also increases memory usage, making this a net negative change.

**Pull request merged**